src/pages/index.js

```diff
@@ -100,36 +146,95 @@ function fillProfileForm() {
     descInput.value = info.description;
 }
 
-function updateInfo(event, data) {
-    userInfo.setUserInfo({
+async function updateInfo(event, data) {
+    const newInfo = {
         name: data["name-input"],
         description: data["description-input"],
+    };
+    userInfo.setUserInfo(newInfo);
+    api.editProfile({
+        name: newInfo.name,
+        about: newInfo.description
     });
 }
 
+function openAvatarModal() {
+    avatarPopup.open();
+    formValidators["avatar-form"].toggleButtonState();
+}
+
+async function updateAvatar(event, data) {
+    const link = data["url-input"];
+    api.updateAvatar(link)
+        .then((res) => {
+            document.querySelector(".profile__image")
+                .setAttribute("src", res.avatar);
+        });
+    
+}
+
 // #endregion 
 
-// #region New place modal & card creation methods
+// #region New place modal & card methods
 
 function openPlaceModal() {
     formValidators["place-form"].toggleButtonState();
     placePopup.open();
 }
 
-function addCard(event, data) {
+function openConfirmModal(event, data) {
+    deletePopup.open();
+    deleteButton.removeEventListener("click", deleteHandler);
+    deleteHandler = () => {
+        deleteCard(event.target.closest(".card"), data);
+    };
+    deleteButton.addEventListener("click", deleteHandler);
+}
+
+async function addCard(event, data) {
     const cardData = {
         name: data["title-input"],
         link: data["url-input"],
     };
-    cardsSection.addItem(getCardElement(cardData));
+
+    api.addCard(cardData)
+        .then((res) => {
+            cardData._id = res._id;
+            cardsSection.addItem(getCardElement(cardData));
+        });
+
     event.target.reset();
```

Comment:
`    event.target.reset();
`

You need to **reset** the form only   in block `then` not to do that immediately after `submit` because when there is an error the popup will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more


**Please, correct it everywhere**